File: listen.py
#!/usr/bin/env python
import pika
import sys
import json

credentials = pika.PlainCredentials('admin', 'admin')

connection = pika.BlockingConnection(pika.ConnectionParameters('130.245.170.104', 5672, '/', credentials))
channel = connection.channel()

channel.exchange_declare(exchange='hw3',
                         exchange_type='direct',
			 durable=True)


result = channel.queue_declare(exclusive=True)
queue_name = result.method.queue
print('queue name: %r' % queue_name)

severities = sys.argv[1:]

if not severities:
    sys.stderr.write("Usage: %s [binding_key]...\n" % sys.argv[0])
    sys.exit(1)

for severity in severities:
    print('bind queue with exchange')
    channel.queue_bind(exchange='hw3',
    		       queue=queue_name,
                       routing_key=severity)

def callback(ch, method, properties, body):
    data = json.loads(body)
    print("msg: {}".format(data['msg']))
    #print(" [x] %r:%r" % (method.routing_key, body))



channel.basic_consume(callback,
                      queue=queue_name,
                      no_ack=True)

print(' [*] Waiting for messages. To exit press CTRL+C')
channel.start_consuming()

File: listen.js
#!/usr/bin/env node
var amqp = require('amqplib/callback_api');
const opt = { credentials: require('amqplib').credentials.plain('admin', 'admin') };

amqp.connect('amqp://130.245.170.104', opt, function(err, conn) {
  conn.createChannel(function(err, ch) {
    var q = 'hello';

    ch.assertQueue(q, {durable: false});
    console.log(" [*] Waiting for messages in %s. To exit press CTRL+C", q);
    ch.consume(q, function(msg) {
      console.log(" [x] Received %s", msg.content.toString());
    }, {noAck: true});
  });
});
